### ISSUE TITLE
Async methods highlight

### ISSUE THREAD

**Issue by RevanScript** [Microsoft/TypeScript-TmLanguage#50]
Async highlight is not working for decorated methods (Atom, atom-typescript 8.1)

```javascript
    @Action('/promise')
    public async asyncResponse(): Promise<Object> {
        return {
            promiseResult: 'promise is ok'
        }
    }
```

also, `class` keyword now looks like `type` in this example:

```javascript
export class IndexController extends AppController {
...
```

original issue: [TypeStrong/atom-typescript#819](https://github.com/TypeStrong/atom-typescript/issues/819)

**Comment by anubmat** [comment]
For part 1, is this being declared outside a class declaration, because the problem does not seem to be 'async', looks like it is with async and public being used outside class definition.
For part 2, Needs more info. It looks good to me now.

**Comment by sheetalkamat** [comment]
Fixed by #257
